### PR TITLE
[netcore] Review and document netcore test failures that are skipped

### PR DESCRIPTION
**Pull request by steveisok**
Went through the majority of corefx tests that were being skipped in CoreFX.issues.rsp.

For each failure, a github issue was logged and if possible, a more expanded
failure reason was specified.

Part of the https://github.com/mono/mono/issues/14787 epic work.

### CODE REVIEW

File: netcore/CoreFX.issues.rsp [hunk 1]
```diff
@@ -73,88 +75,132 @@
...
 
-# Depends on precise GC
--nomethod System.IO.Tests.FileStream_Dispose.FinalizeFlushesWriteBuffer
--nomethod System.IO.Tests.FileStream_Dispose.NoDispose_CallsVirtualDisposeFalseArg_ThrowsDuringFlushWriteBuffer_FinalizerWontThrow
 # TODO: Hangs
 -nomethod System.IO.Tests.Directory_SetCurrentDirectory+Directory_SetCurrentDirectory_SymLink.SetToPathContainingSymLink
 
 ####################################################################
 ##  System.IO.Tests
 ####################################################################
```
Comment (filipnavara) [inline netcore/CoreFX.issues.rsp]: This header should be removed too, the section is empty now.

File: netcore/CoreFX.issues.rsp [hunk 2]
```diff
@@ -202,13 +268,19 @@
...
+
 # TODO: Hangs
+# https://github.com/mono/mono/issues/15006
 -nomethod *ExpectedDiagnosticSourceActivityLogging*
 
 # Works, but may trigger UI!
--nomethod System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test.SetDelegate_ConnectionSucceeds
--nomethod System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_ClientCertificates_Test.AutomaticOrManual_DoesntFailRegardlessOfWhetherClientCertsAreAvailable
+# NOTE: KEEPING THIS HERE BUT COMMENTED OUT - in the event the UI is shown (wasn't for me)
+#-nomethod System.Net.Http.Functional.Tests.SocketsHttpHandler_HttpClientHandler_DangerousAcceptAllCertificatesValidator_Test.SetDelegate_ConnectionSucceeds
```
Comment (filipnavara) [inline netcore/CoreFX.issues.rsp]: I will re-run on my machine to check.

**Comment by steveisok** [comment]
@monojenkins build

**Comment by marek-safar** [comment]
@steveisok there are new failing tests, is it expected?

**Comment by marek-safar** [comment]
@monojenkins build

**Comment by steveisok** [comment]
@monojenkins build failed

**Comment by steveisok** [comment]
@monojenkins build Windows i386

**Pull request merged**